### PR TITLE
[PROD-1044] Refactor and CLI update for easier Airflow integration

### PR DESCRIPTION
**Pull request by rmoneys**
https://synccomputing.atlassian.net/browse/PROD-1044

Pull out cluster update logic to the developer interface so that developers of the library don't have to duplicate it.

Also add configure command options to support unattended installation.

### CODE REVIEW

File: sync/cli/__init__.py [hunk 1]
```diff
@@ -33,40 +33,57 @@ def main(debug: bool):
 
 
 @main.command
-def configure():
+@click.option("--api-key-id")
+@click.option("--api-key-secret")
+@click.option("--prediction-preference")
+@click.option("--databricks-host")
+@click.option("--databricks-token")
+@click.option("--databricks-region")
```
Comment (rmoneys) [inline sync/cli/__init__.py:41]: If users provide all options they won't be prompted

File: sync/awsdatabricks.py [hunk 2]
```diff
@@ -416,10 +402,51 @@ def get_prediction_job(
                         if j.get("job_cluster_key") != cluster_key
                     ] + [{"job_cluster_key": cluster_key, "new_cluster": prediction_cluster}]
                 else:
+                    # For `new_cluster` definitions, Databricks will automatically assign the newly created cluster a name,
+                    # and will reject any run submissions where the `cluster_name` is pre-populated
+                    if "cluster_name" in prediction_cluster:
+                        del prediction_cluster["cluster_name"]
```
Comment (rmoneys) [inline sync/awsdatabricks.py:408]: This is also in `run_job_object` but it doesn't hurt to have it here too

File: sync/awsdatabricks.py [hunk 3]
```diff
@@ -416,10 +402,51 @@ def get_prediction_job(
...
-            return cluster_response
-        return Response(error=DatabricksError(message="No task found in job"))
+            return prediction_cluster_response
+        return cluster_response
+    return Response(error=DatabricksError(message="No task found in job"))
+
+
+def get_prediction_cluster(
+    cluster: dict, prediction_id: str, preference: str = CONFIG.default_prediction_preference.value
+) -> Response[dict]:
```
Comment (gseyffert-sync) [inline sync/awsdatabricks.py:418]: :+1:

**Review by gseyffert-sync — DISMISSED** [review]
Makes sense to me :+1: thanks!

**Review by gseyffert-sync — APPROVED** [review]

**Review by gseyffert-sync — APPROVED** [review]

**Pull request merged**